packages/@expo/prebuild-config/src/plugins/icons/AssetContents.ts

```diff
@@ -64,9 +64,11 @@ export async function writeContentsJsonAsync(
   directory: string,
   { images }: Pick<ContentsJson, 'images'>
 ): Promise<void> {
-  await fs.ensureDir(directory);
+  if (!fs.existsSync(directory)) {
+    await fs.promises.mkdir(directory, { recursive: true });
+  }
```

Comment:
```suggestion
  await fs.promises.mkdir(directory, { recursive: true });
```

nit: quick question. for ensureDir migration in other places, we didn't check existsSync. can we just mkdir?

Reply:
Yep, that was a bit of a random choice on my end to match what `fs-extra` does more closely, but I can get those removed 👍 I'll make sure to organise that into the git history, rather than committing on top to keep this more reviewable